### PR TITLE
doc: Improve IC guide in PR Template

### PR DESCRIPTION
**Pull request by sravan-s**
Improve PR template to clarify more about conventional commits
Fixes: https://sendbird.atlassian.net/browse/UIKIT-4143

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
+## For Internal Contributors
+
+* Follow trunk workflow(https://trunkbaseddevelopment.com/)
```
Comment (sravan-s) [inline .github/pull_request_template.md:3]: For normal situations, lets follow trunk workflow..
But if there is a major change - say, rewrite of SDK or major version update such as v4 -> v5..
We can choose other flows

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+Add a brief description of the changes that you have involved in this PR
+Bullet points are also fine
+
+// Section: Footer. Recommended!
+// Recommended! if you are dealing with SB_ISSUES(Customer issues) or release tickets
+Fixes: [<TICKET_ID>](https://sendbird.atlassian.net/browse/<TICKET_ID>)
+
+// If you want to add coauthors - say you pair programmed or they made significant contributions to the ideas in code
+// and you want to thank them
+Co-authored-by: name <name@example.com>, name2 <name@example.com>
```
Comment (sravan-s) [inline .github/pull_request_template.md:29]: This is to follow -> https://git-scm.com/docs/git-interpret-trailers 
They come handy when it comes to autogenerating changelog etc
https://zerokspot.com/weblog/2020/10/24/git-commit-messages-with-attributes/

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+* Branch naming -> `{type}/TICKET_ID/description`
+  * Where type = `feat` / `feature` / `fix` / `chore` / `doc` / `release`
+* Always recive PR review approvals
+* Rebase your branch with main, wait for CI to pass
+* Squash merge your commit
+  * Follow the following template for both PR description and squashing to main
+  * Use imperative language in title and description
+  * The template is very similar to conventional commits. [Read More](https://www.conventionalcommits.org/en/v1.0.0/)
+
+Template
```
Comment (sravan-s) [inline .github/pull_request_template.md:13]: A lot of thought leaders advocate for 50/72 formatting
ie: max 50 chars for PR title, and 72 for summary. (not a hard limit)
Its good to follow those ~ especially for people primarily using command line ->
max 50 chars makes these jobs more easy to do through command line:
* Rebasing
* Cherry-picking
* git log --oneline
* Git lens in your ide
 
https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Review by sravan-s — COMMENTED** [review]
Overall, the point is that, this is an open project and 
a lot of customers fork it. They wont have link to our JIRA tickets

Also, if it saves our engineer 5 mins(ie - not have to read a jira ticket),
to understand why an issue/code-change happened, through commit messages,
or  git lens, Its a win!

This PR is to talk about commit history, we have other intiatives about code quality
through lint and a testing initiative in place ~

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,3 +1,33 @@
...
+
+* Follow trunk workflow(https://trunkbaseddevelopment.com/)
+* Branch naming -> `{type}/TICKET_ID/description`
+  * Where type = `feat` / `feature` / `fix` / `chore` / `doc` / `release`
+* Always recive PR review approvals
+* Rebase your branch with main, wait for CI to pass
+* Squash merge your commit
+  * Follow the following template for both PR description and squashing to main
+  * Use imperative language in title and description
+  * The template is very similar to conventional commits. [Read More](https://www.conventionalcommits.org/en/v1.0.0/)
```
Comment (AhyoungRyu) [inline .github/pull_request_template.md:11]: nit suggestion; just made some little touch ✏️ 
```suggestion
* Follow the trunk workflow(https://trunkbaseddevelopment.com/)
* Branch naming format -> `{type}/TICKET_ID/description`
  * Where type = `feat` / `feature` / `fix` / `chore` / `doc` / `release`
* Always receive PR review approvals
* Rebase your branch with the main branch, and wait for CI to pass
* Squash merge your commit
  * Follow the provided template for both the PR description and squashing to the main branch
  * Use imperative language in the title and description
  * The template is very similar to conventional commits. [Read More](https://www.conventionalcommits.org/en/v1.0.0/)
```

File: .github/pull_request_template.md (hunk 2 above)
Comment (AhyoungRyu) [inline .github/pull_request_template.md:29]:
```suggestion
// PR title (Required)
[type]: A short description of the changes in imperative language.

// Section: PR description (Optional)
Add a brief description of the changes that you have involved in this PR.
Bullet points are also fine.

// Section: Footer (Recommended)
// Recommended if you are dealing with SB_ISSUES (Customer issues) or release tickets.
Fixes: [<TICKET_ID>](https://sendbird.atlassian.net/browse/<TICKET_ID>)

// If you want to add coauthors - for example, if you pair programmed or they made significant contributions to the ideas in the code and you want to thank them.
Co-authored-by: Name <name@example.com>, Name2 <name@example.com>
```

**Review by AhyoungRyu — APPROVED** [review]
Do u want to have more discussion about this? I'm good with the overall change :)
I just left some small suggestions on wordings.

**Comment by sravan-s** [comment]
@AhyoungRyu commited your changes

**Comment by AhyoungRyu** [comment]
I'm good with the change. Feel free to merge when you think it's okay to go.

**Pull request merged**